### PR TITLE
merge libsql

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: cmd/atlas/internal/sqlparse/sqlparse.go [hunk 1]
```diff
@@ -52,5 +52,4 @@ func init() {
 	Register(mysql.DriverName, &myparse.Parser{})
 	Register(postgres.DriverName, &pgparse.Parser{})
 	Register(sqlite.DriverName, &sqliteparse.FileParser{})
-	Register(sqlite.LibsqlDriverName, &sqliteparse.FileParser{})
```
Comment (haaawk) [inline cmd/atlas/internal/sqlparse/sqlparse.go]: Isn't this needed @a8m ?

File: sql/sqlite/driver.go [hunk 2]
```diff
@@ -60,6 +60,16 @@ func init() {
 			return uc
 		})),
 	)
+	sqlclient.Register(
+		"libsql",
+		sqlclient.DriverOpener(Open),
+		sqlclient.RegisterTxOpener(OpenTx),
+		sqlclient.RegisterCodec(MarshalHCL, EvalHCL),
+		sqlclient.RegisterFlavours("libsql+wss", "libsql+https"),
```
Comment (haaawk) [inline sql/sqlite/driver.go:68]: At the moment +https won't work as it does not support interactive transactions and is missing some features. We're working on making it better but maybe it's worth to keep https out of scope for this PR?

File: doc/md/concepts/url.mdx [hunk 3]
```diff
@@ -101,6 +101,14 @@ Connecting to an in-memory SQLite database (ephemeral). Useful for `--dev-url`:
 sqlite://file?mode=memory&_fk=1
 ```
 
+Atlas also supports WebSocket and HTTPS connections to remote `libsql` databases:
+
+```shell
+libsql+wss://database-url
+
+libsql+https://database-url
```
Comment (haaawk) [inline doc/md/concepts/url.mdx:109]: https support is under development so it might be a good idea to keep it out of scope of this PR

**Review by haaawk — APPROVED** [review]
I've tested the PR against Turso (managed Libsql) and updated https://github.com/chiselstrike/atlas-integration-ci to use the `haaawk-libsql` branch for testing. All tests are passing.

File: cmd/atlas/internal/sqlparse/sqlparse.go (hunk 1 above)
Comment (a8m) [reply cmd/atlas/internal/sqlparse/sqlparse.go]: No. For `migrate lint` we recommend users to pass a URL to an in-memory SQLite database. The semantics are the same

**Pull request merged**